### DISCUSSION TITLE
Restyling `TimePicker` popup position

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#13692] [Q&A]
Hi, I tried to restyle `TimePicker`, but I couldn't because the popup location is hardwired into code(#7634).

https://github.com/AvaloniaUI/Avalonia/blob/1a798ff44cd2e976cfb828aaf3f8d8e790d811eb/src/Avalonia.Controls/DateTimePickers/TimePicker.cs#L255-L271

My question, if it is by design.

**Comment by timunie** [comment]
I think it is not by design. But back when it was written, the best we had. These controls would really need an overhauling :-)

**Reply by workgroupengineering** [reply]
Can I make  a PR?

**Reply by timunie** [reply]
From my point of view I see no reason why not. I feel 95% confident it will be merged. Unless I oversee smth here 👓
